File: src/main/java/com/api/prisma_vi/user/UsersRepository.java
package com.api.prisma_vi.user;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Repository;

@Repository
public interface UsersRepository extends JpaRepository<Users, Long> {

     UserDetails findByEmail(String email);

     Boolean existsByEmail(String email);
}

File: src/main/java/com/api/prisma_vi/user/UsersRepository.kt
package com.api.prisma_vi.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Repository

@Repository
interface UsersRepository : JpaRepository<Users, Long> {

    fun findByNickName(nickName: String): UserDetails
}